components/AutoComplete/AutoComplete.jsx

```diff
@@ -0,0 +1,381 @@
+    if (filteredValues.length === 0) {
+      setShowSuggestions(false);
+    } else {
+      setShowSuggestions(true);
+    }
+    setFilterSuggestions(filteredValues);
+    setFilterSuggestionsLength(filteredValues.length);
+  };
+
+  const handleChange = ({ target }) => {
+    setUserTypedValue(target.value);
+    if (selectedItem) selectedItem(target.value);
+    setCursor(0);
+    handleFilter(target.value);
+  };
+
+  /* istanbul ignore next */
+  const handleClickOutside = event => {
+    if (!wrapperRef.current.contains(event.target)) {
+      setShowSuggestions(false);
+    }
+  };
+
+  const handleInputClick = () => {
+    const filteredItem = filterItens(userTypedValue);
+    setFilterSuggestions(filteredItem);
+    setFilterSuggestionsLength(filteredItem.length);
+    setShowSuggestions(true);
+  };
+
+  const handleItemClick = item => {
+    setUserTypedValue(item);
+    if (selectedItem) selectedItem(item);
+    setShowSuggestions(false);
+  };
+
+  const handleClearValue = () => {
+    setUserTypedValue('');
+  };
+
+  /* istanbul ignore next */
+  const handleEscPress = ({ key }) => {
+    if (key === EscapeKeyPressValue) {
+      setShowSuggestions(false);
+      wrapperRef.current.children[1].focus();
+      setCursor(0);
+    }
+  };
+
+  const generateSuggestions = () =>
+    showSuggestions && filterSuggestions ? (
+      <List
+        theme={theme}
+        role="listbox"
+        id="autocompleteOptions"
+        ref={listOptions}
+      >
+        {filterSuggestions.map((item, index) => (
+          <ListItens
+            key={item}
+            aria-posinset={index}
+            onClick={() => handleItemClick(item)}
+            theme={theme}
+            aria-selected={index === cursor}
+            role="option"
+            tabIndex="-1"
+          >
+            {item}
+          </ListItens>
+        ))}
+      </List>
+    ) : null;
+
+  const generateAssistiveDescript = () => {
+    if (showSuggestions) {
+      return assistiveDescriptionDropDownOpen;
+    }
+    return assistiveDescriptionDefault;
+  };
+
+  const downPress = useKeyPress('ArrowDown');
+  const upPress = useKeyPress('ArrowUp');
+  const enterPress = useKeyPress('Enter');
+  const tabPress = useKeyPress('Tab');
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (showSuggestions && filterSuggestionsLength && downPress) {
+      const selectedCursor =
+        cursor < filterSuggestionsLength - 1 ? cursor + 1 : cursor;
+      setCursor(selectedCursor);
+      listOptions.current.children[selectedCursor].focus();
+    }
+  }, [downPress]);
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (showSuggestions && filterSuggestionsLength && upPress) {
+      const selectedCursor = cursor > 0 ? cursor - 1 : cursor;
+      setCursor(selectedCursor);
+      listOptions.current.children[selectedCursor].focus();
+    }
+    if (upPress && cursor === 0) {
+      wrapperRef.current.children[1].focus();
+      setCursor(-1);
```

Comment:
When I navigate using arrow up in options this error occurs:

![image](https://user-images.githubusercontent.com/82893321/148788999-2efe6387-a5aa-4b42-9567-a0c7d46360bf.png)

Reply:
Can you show me this later? Didn't manage to reproduce